### PR TITLE
[#IR-215] Show the link to task in the issue tracker on the issue summary and retrospect task modal

### PR DESCRIPTION
**Pull request by himanshu-garg-165**

### CODE REVIEW

File: src/assets/stylesheets/tables.scss [hunk 1]
```diff
@@ -133,3 +133,14 @@ mat-row {
         padding: 4px !important;
     }
 }
+
+.custom-ag-grid-anchor-cell {
```
Comment (anmolghosh) [inline src/assets/stylesheets/tables.scss]: - update pointer
- see if we could have any better name for the class

File: src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts [hunk 2]
```diff
@@ -325,6 +333,9 @@ export class SprintTaskSummaryComponent implements OnInit, OnChanges, OnDestroy
                     newRowsAction: 'keep',
                     clearButton: true,
                 },
+                onCellClicked: (event: CellClickedEvent) => {
+                    window.open(event.data.Url);
```
Comment (anmolghosh) [inline src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts]: add settings to open in new tab

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.scss [hunk 3]
```diff
@@ -30,3 +32,14 @@
     float: left;
     clear: both;
 }
+
+a.task-link {
+    color: $light-primary-text;
+    display: inline-block;
+    text-decoration: none;
```
Comment (anmolghosh) [inline src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.scss]: check pointer

File: src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts [hunk 4]
```diff
@@ -325,6 +327,12 @@ export class SprintTaskSummaryComponent implements OnInit, OnChanges, OnDestroy
                     newRowsAction: 'keep',
                     clearButton: true,
                 },
+                cellRenderer: (params) => {
+                    if (!params.data.IsTrackerTask || !params.data.URL) {
```
Comment (anmolghosh) [inline src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts]: could we use lodash here?

File: src/assets/stylesheets/tables.scss [hunk 5]
```diff
@@ -41,6 +41,18 @@ mat-row {
             border-color: #3f51b5 !important;
         }
     }
+
+    &-anchor-cell {
```
Comment (anmolghosh) [inline src/assets/stylesheets/tables.scss]: why are we using `&` in name of class?

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.html [hunk 6]
```diff
@@ -1,6 +1,17 @@
 <div class="retrospect-task app-dialog">
     <h2 mat-dialog-title>
-        {{taskDetails.Key}}<span *ngIf="taskDetails.Summary"> : {{taskDetails.Summary}}</span>
+        <span *ngIf="taskDetails.URL; then taskUrl else taskKey"></span>
+        <ng-template #taskUrl>
```
Comment (anmolghosh) [inline src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.html]: `taskUrlView`

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.html [hunk 7]
```diff
@@ -1,6 +1,17 @@
...
+            <a class="task-link"
+               [href]="taskDetails.URL"
+               target="_blank">
+                {{taskDetails.Key}}
+            </a>
+        </ng-template>
+        <ng-template #taskKey>
```
Comment (anmolghosh) [inline src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.html]: `taskKeyView`

File: src/assets/stylesheets/tables.scss (hunk 5 above)
Comment (himanshu-garg-165) [reply src/assets/stylesheets/tables.scss]: It is a way to name classes. Refer https://css-tricks.com/the-sass-ampersand/#article-header-id-12

File: src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts (hunk 4 above)
Comment (himanshu-garg-165) [reply src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts]: There is no direct method in lodash to check for multiple keys simultaneously as of now, but there is one workaround for that, which is a little less readable. Refer https://stackoverflow.com/questions/29001762/lodash-has-for-multiple-keys

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.scss (hunk 3 above)
Comment (himanshu-garg-165) [reply src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.scss]: Done

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.html (hunk 6 above)
Comment (himanshu-garg-165) [reply src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.html]: Done

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.html (hunk 7 above)
Comment (himanshu-garg-165) [reply src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.html]: Done

**Pull request merged**